app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -571,6 +575,48 @@ public class MainActivity : AppCompatActivity(), MainViewController, RouteCallba
         hideProgress()
     }
 
+    private fun zoomToShowRoute(route: Route) {
+        //Get size of display For calculations
+        var mdisp = getWindowManager().getDefaultDisplay()
+        var mdispSize = Point()
+        mdisp.getSize(mdispSize)
+        var width = mdispSize.y
+
+        var geometry = route?.getGeometry()
+        var start = geometry?.get(0)
+        var finish = geometry?.get(geometry.size - 1)
+        var distance = finish?.distanceTo(start)
+
+        var equatorCircumfrence: Double = 40075.16
+
+        var lat: Double = (finish?.latitude  + start?.latitude) / 2.0
+        var lon: Double = (finish?.longitude + start?.longitude) / 2.0
+        mapController?.setMapPosition(lon, lat)
+
+        var numPixelsRequired = distance?.toFloat() / width.toFloat()
+        var zoomLevel = Math.log(((equatorCircumfrence * Math.abs(Math.cos(lat)))
+                / (numPixelsRequired?.toDouble()))) / Math.log(2.0)
+
+        setMapRotation(0f)
+        setMapTilt(0.0f)
+        mapController?.setMapZoom(zoomLevel.toFloat() * .8f)
```

Comment:
Currently the map seems a little too zoomed out in route preview on my device (Nexus 5). Can we try `0.9f` scale factor here?